app/containers/Menu.js

```diff
@@ -0,0 +1,254 @@
+    this.state.drinkList.push({name: this.state.drinkName, ingredients: this.state.newdrinkingredients, rating: 5, drinkNumber: this.state.drinkList.length, url: 'http://google.com/#q=' + urlName + '+cocktail'});
+    this.setState({
+      drinklist: this.state.drinkList,
+      drinkName: '',
+      newdrinkingredients: '',
+    });
+  } else {
+    this.setState({
+      drinkName: '',
+      newdrinkingredients: '',
+    });
+  }
+
+  },
+
+  handlePopSort: function () {
+    this.setState({
+      drinkList: this.popularSort(),
+    });
+  },
+
+  handleAlphSort: function () {
+    this.setState({
+      drinkList: this.alphabeticalSort(),
+    });
+  },
+
+  removeDrink: function (name) {
+    var index = this.state.drinkList.indexOf(name);
+    this.state.drinkList.splice(index, 1);
+    this.setState({
+      drinkList: this.state.drinkList,
+    });
+  },
+
+  editName: function (e) {
+    this.setState({
+      drinkList: e.target.value,
+    });
+    console.log(this.state.editDrinkName);
+    /*var index = this.state.drinkList.indexOf(name);
+    this.state.drinkList[index].name = this.state.editDrinkName;
+    this.setState({
+      drinkList: this.state.drinkList,
+    });*/
+
+  },
+
+  editDrinkList: function (e) {
+    console.log(this.state.drinkList.indexOf(e));
+  },
+
+  /*editIngredients: function (name) {
+    console.log(this.state.editIngredients);
+    var index = this.state.drinkList.indexOf(name);
+    this.state.drinkList[index].ingredients = this.state.EditDrinkIngredients;
+    this.setState({
+      drinkList: this.state.drinkList,
+    });
+  },*/
+
+  drinkListReturn: function () {
+  if (this.state.isEditing === 'Save') {
+    return this.state.drinkList.map(function (drink, index) {
+      return (
+        <Editor object={drink} index={index}/>
+        );
+    });
+  } else {
+    return this.state.drinkList.map(function (index) {
+      return (
+        <div style={{color: 'black'}}>
+          <li>
+            <div>
+            <button type='button' className='btn btn-danger' style={{float: "right"}} onClick={function() {this.removeDrink(index)}.bind(this)}> X </button>
+            <a href={index.url} style={{color: 'black'}}>
+              <h4 key={index}>{index.name}</h4>
+              <p>{index.ingredients}</p>
+            </a>
+            </div>
+          </li>
+        </div>
+      );
+    }.bind(this)
+    );
+  };
+  },
+
+  popularSort: function () {
+    this.state.drinkList.sort(function (drink1, drink2) {
+      return drink2.rating - drink1.rating;
+    });
+    this.setSortingPopularFn();
+    return this.state.drinkList;
+  },
+
+  alphabeticalSort: function () {
+    this.state.drinkList.sort(function (index1, index2) {
+        var drink1 = index1.name.toUpperCase();
+        var drink2 = index2.name.toUpperCase();
+        if (drink1 < drink2) {
+          return -1;
+        }
+        if (drink1 > drink2) {
+          return 1;
+        }
```

Comment:
kewl use of ternarys could make this more readable.  

```
if (drink1 == drink2) return 0;
return (drink1 < drink2) ? -1 : 1;
```
[reference on terneries from SO](http://stackoverflow.com/a/6260001)